pkg/operator/ceohelpers/bootstrap.go

```diff
@@ -107,13 +107,18 @@ func CheckSafeToScaleCluster(
 	infraLister configv1listers.InfrastructureLister,
 	etcdClient etcdcli.AllMemberLister) error {
 
-	bootstrapComplete, err := IsBootstrapComplete(configmapLister, staticPodClient, etcdClient)
+	bootstrapComplete, err := IsBootstrapComplete(configmapLister, etcdClient)
 	if err != nil {
 		return fmt.Errorf("CheckSafeToScaleCluster failed to determine bootstrap status: %w", err)
 	}
 
-	// while bootstrapping, scaling should be considered safe always
-	if !bootstrapComplete {
+	revisionStable, err := IsRevisionStable(staticPodClient)
+	if err != nil {
+		return fmt.Errorf("CheckSafeToScaleCluster failed to determine stability of revisions: %w", err)
+	}
+
+	// while bootstrapping and when revision is stabilising, scaling should be considered safe always
+	if !bootstrapComplete || !revisionStable {
```

Comment:
I know it's not an expensive call, but feel free to early exit on the first check already. There's no sense in checking revision stability if the bootstrap is not completed yet